core/src/main/java/com/google/errorprone/bugpatterns/UnnecessarilyFullyQualified.java

```diff
@@ -169,12 +170,16 @@ public Void visitIdentifier(IdentifierTree identifierTree, Void aVoid) {
         continue;
       }
       List<TreePath> pathsToFix = getOnlyElement(types.values());
-      if (pathsToFix.stream()
-          .anyMatch(path -> findIdent(nameString, state.withPath(path), VAL_TYP) != null)) {
+      if (!pathsToFix.get(0).getLeaf().toString().contains(JAVA_LANG)
+          && pathsToFix.stream()
+              .anyMatch(path -> findIdent(nameString, state.withPath(path), VAL_TYP) != null)) {
         continue;
       }
```

Comment:
I don't think we should to special-case `java.lang` here. Instead we should generalize the `anyMatch` logic and compare any matching ident again the type symbol under consideration; if they're the same we can proceed.
```suggestion
      if (pathsToFix.stream()
          .anyMatch(
              path -> {
                Symbol ident = findIdent(nameString, state.withPath(path), VAL_TYP);
                return ident != null && !ident.equals(type);
              })) {
        continue;
      }
```